lib/api/graphql/graphql_client_provider.dart

```diff
@@ -8,35 +15,72 @@ class GraphQLClientProvider {
   const GraphQLClientProvider._(this._httpLink);
 
   static AuthLink _authLink;
+  static CustomRefreshLink _refreshLink;
+  static GraphQLClient _client;
 
   factory GraphQLClientProvider() {
     return GraphQLClientProvider._(
       HttpLink(F.graphQLEndpoint),
     );
   }
 
-  // Setup a token to all request i.e: 'Bearer - token'
+  GraphQLClient get client {
+    if (_client == null) {
+      Link _link;
+      if (_authLink != null) {
+        _link = Link.from([_authLink, _refreshLink, _httpLink]);
+      } else {
+        _link = _httpLink;
+      }
+      _client = GraphQLClient(
+        cache: GraphQLCache(),
+        link: _link,
+      );
+    }
+    return _client;
+  }
+
+  /// Setup a token to all request i.e: 'Bearer - token'
+  /// Also set up the refresh token [Link]
   void setAuthToken(String token) {
     _authLink = AuthLink(
       getToken: () async => token,
     );
+
+    _refreshLink = CustomRefreshLink.oAuth2(
+      tokenStorage: LocalSharedPreferencesStorage(),
+      doRefreshing: (oauth2Token) async {
+        final httpClient = Get.find<ApiClient>();
+        final refreshResult = await httpClient.refreshToken(
+          RefreshTokenRequest(
+            refreshToken: oauth2Token.refreshToken,
+            clientId: F.basicAuthClientId,
+            clientSecret: F.basicAuthClientSecret,
+          ),
+        );
+
+        if (refreshResult.data != null) {
+          final tokenResponse = refreshResult.data.authToken;
+          return OAuth2Token(
+            accessToken: tokenResponse.accessToken,
+            refreshToken: tokenResponse.refreshToken,
+            expiresIn: tokenResponse.expiresIn,
+            tokenType: tokenResponse.tokenType,
+          );
+        } else {
+          return null;
+        }
```

Comment:
What do think about this suggestion? It has better readable in IMO 🤔 
```suggestion
        if (refreshResult.data == null) {
          return null;
        }
        
         final tokenResponse = refreshResult.data.authToken;
         return OAuth2Token(
            accessToken: tokenResponse.accessToken,
            refreshToken: tokenResponse.refreshToken,
            expiresIn: tokenResponse.expiresIn,
            tokenType: tokenResponse.tokenType,
          );
```